route_following_plugin/src/route_following_plugin.cpp

```diff
@@ -98,32 +90,31 @@ namespace route_following_plugin
         }
         double current_progress = wm_->routeTrackPos(current_loc).downtrack;
         double speed_progress = current_speed_;
-        double total_maneuver_length = current_progress + mvr_duration_ * RouteFollowingPlugin::FIFTEEN_MPH_IN_MS;
+
+        double target_speed=findSpeedLimit(current_lanelet);   //get Speed Limit
+
+        double total_maneuver_length = current_progress + mvr_duration_ * target_speed;
         
         while(current_progress < total_maneuver_length && last_lanelet_index < shortest_path.size())
         {
             ROS_ERROR_STREAM("Lanlet: " << shortest_path[last_lanelet_index].id());
```

Comment:
It seems like these ERROR messages for debug purposes have been overlooked by last PR.
Please change them to DEBUG